### DISCUSSION TITLE
Failed to push or attach annotation file

### DISCUSSION THREAD

**Discussion by FeynmanZhou** [oras-project/oras#664] [Q&A] [answered]
Hi team,

I would like to use ORAS 0.15.1 to attach or push an annotation file `test.json` to an artifact:

```
cat test.json
{
"name":"John"
"name":"Bob"
}
```

But I failed to attach or push it and got the following error info:

```
oras push --annotation-file test.json localhost:5000/hello:latest
Error: json: cannot unmarshal string into Go value of type map[string]map[string]string
```

```
oras attach --artifact-type doc/example --annotation-file test.json feynmanacr.azurecr.io/sample/hello-world:latest test.json
Error: json: cannot unmarshal string into Go value of type map[string]map[string]string
```

Is any requirements or prerequisites I missed?

**Comment by qweeah** [comment, marked as answer]
The format of manifest annotation file `test.json` is not right, the right format is displayed as below:
```
{
  ...
  "<filename>": {
    "<annotation_key>": "annotation_value"
  }
}
```
where `<filename>` can be `$manifest`, `$config` or name of any to-be-uploaded files.

For example, if you need to add `"name":"John"` to pushed manifest, then `test.json` should be like:
```
{"$manifest":{"name":"John"}}
```

Detailed doc: https://oras.land/cli/4_manifest_annotations/
